### DISCUSSION TITLE
Pearson Coefficient?

### DISCUSSION THREAD

**Discussion by athenawisdoms** [Lightning-AI/torchmetrics#153] [Regression] [answered]
Hi, is it possible to calculate the Pearson Coefficient metric using torchmetrics?

**Comment by SkafteNicki** [comment, marked as answer]
It is currently not possible, but it definitely is a metric we should have in our collection.
I can take a look at implementing it one of the following days.

Also, would spearman's rank correlation coefficient be of any value for you?

**Reply by athenawisdoms** [reply]
Awesome thank you!

Yes, that's a great point. Spearman's rank correlation will be very useful as well if it can also be included in TorchMetrics!

**Reply by SkafteNicki** [reply]
I have opened the following two PRs implementing pearson and spearman coefficient:
https://github.com/PyTorchLightning/metrics/pull/157
https://github.com/PyTorchLightning/metrics/pull/158
I cannot guaranty when they are merged, but I will let you know when it happens
